src/data/finsupp/basic.lean

```diff
@@ -1825,6 +1829,35 @@ begin
     apply h_cases (hl $ finset.mem_coe.2 $ mem_support_iff.2 $ λ h, h_contr h.symm) }
 end
 
+variable {f}
+
+@[simp] lemma comap_domain_zero :
+  comap_domain f (0 : β →₀ M) (hf.inj_on _) = (0 : α →₀ M) :=
+by { ext, simp only [comap_domain_apply, coe_zero, pi.zero_apply] }
+
+@[simp]
+lemma comap_domain_single {a : α} {m : M} :
+  comap_domain f (finsupp.single (f a) m) (hf.inj_on _) = finsupp.single a m :=
```

Comment:
As above, this should take a weaker `hf`

Reply:
I am tempted to use `f \inv {f a}`, which technically is too much, since, when `m = 0`, `single _ 0` has empty support.  Does this slightly stronger assumption sound reasonable?

Reply:
Ok, the proof can likely be golfed.  Is this what you had in mind?